### PR TITLE
Add support for version 1.7, add support for testing role with molecule

### PR DESCRIPTION
**Pull request by haad**
ansible test framework, split haproxy.cfg into files per haproxy_listen/backend/frontend definition.

### CODE REVIEW

**Comment by tersmitten** [comment]
Version 1.7 already works

**Comment by tersmitten** [comment]
Does the `split haproxy.cfg into files` work on all versions?

**Comment by tersmitten** [comment]
Why is a new `/etc/init.d/haproxy` needed?

**Comment by tersmitten** [comment]
I'm really happy with a working `molecule` test suite (good work!), but I don't want it to replace the current tests. As I understand, `molecule` does not work for older versions of Ansible. Maybe you can add a new playbook in the `tests` folder (`molecule.yml`)?

**Comment by haad** [comment]
Split to multiple files is done by creating multiple files in /etc/haproxy/conf.d representing different frontends/backends/listens that should work on all versions of haproxy. Difference is that on 1.7 you can point haproxy to /etc/haproxy/conf.d and it will load all files in it. In older versions you have to use multiple -f options. New init.d scripts adds another -f option pointing to directory, but I can change it to point to dir if 1.7 is used or name all files with -f arguments.

Molecule supports ansible version > 1.9.6 which is last supported version. But I can rename molecule test to molecule.yml for sure.

**Comment by haad** [comment]
I have checked why travis tests failed it could work with them if we would keep ansible.cfg in main directory. 

```
[defaults]
roles_path = ../
verbose = vvv

display_skipped_hosts=0
host_key_checking = False
retry_files_enabled = False
```

Basically we need to add parent_directory where checkouted role is into roles_path.

**Comment by tersmitten** [comment]
I rather not use `molecule` in travis-ci (yet).

**Comment by tersmitten** [comment]
> New init.d scripts adds another -f option pointing to directory, but I can change it to point to dir if 1.7 is used or name all files with -f arguments

I rather not use custom init scripts unless it's really needed. What about 16.04 for instance?

**Comment by haad** [comment]
You don't need to use molecule there, just reuse same playbook for both which will save you time.

**Comment by haad** [comment]
So reason to split haproxy configuration into multiple files(for me) is to make sure we can split configuration definition into multiple parts. During deployment where we use this role we deploy loadbalancing config from 3 playbooks with different variables. Before they were just overwriting one file, now they will deploy different files which can be used after restart

**Comment by haad** [comment]
I have moved molecule tests to tests/molecule and removed init script template as it's not needed.

**Comment by haad** [comment]
If we want to make this compatible with older versions of haproxy (1.5,1.6) we would need to use some mechanism like this 

```yaml
    - set_fact:
        haproxy_configs: "{% for config in haproxy_listen %}-f listen-{{ config.name }}.cfg {% endfor %}
                  {% for config in haproxy_backend %}-f backend-{{ config.name }}.cfg {% endfor %}
                  {% for config in haproxy_frontend %}-f frontend-{{ config.name }}.cfg {% endfor %}"

    - debug: var=haproxy_configs
```

This ugly set_fact will create a variable which will contain entries based on frontend/backend/listen

```console
PLAY [localhost] ***************************************************************

TASK [set_fact] ****************************************************************
ok: [localhost]

TASK [debug] *******************************************************************
ok: [localhost] => {
    "haproxy_configs": "-f listen-stats.cfg -f listen-mysql.cfg  -f backend-webservers.cfg  -f frontend-http.cfg "
}

PLAY RECAP *********************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0
```

File: tasks/install.yml [hunk 1]
```diff
@@ -11,14 +11,8 @@
   apt:
     name: "{{ item.name }}"
     state: "{{ item.state }}"
-  with_items: "{{ haproxy_dependencies }}"
+  with_items:
+    - "{{ haproxy_dependencies }}"
+    - "{{ haproxy_install | default('latest') }}"
```
Comment (tersmitten) [inline tasks/install.yml]: This seems incorrect

Comment (tersmitten) [reply tasks/install.yml]: Ans both `haproxy_dependencies` and `haproxy_install` are lists, so yuou probably need to do something like `haproxy_dependencies + haproxy_install`

**Comment by tersmitten** [comment]
Closing this due to inactivity

**Pull request closed without merging**